tests/bcl-test/templates/iOSApp/ViewController.cs

```diff
@@ -99,7 +99,7 @@ public async override void ViewDidLoad ()
 			
 			logger.Info ($"Tests run: {runner.TotalTests} Passed: {runner.PassedTests} Inconclusive: {runner.InconclusiveTests} Failed: {runner.FailedTests} Ignored: {runner.FilteredTests}");
 			if (options.TerminateAfterExecution)
-				TerminateWithSuccess ();
+				BeginInvokeOnMainThread (TerminateWithSuccess);
```

Comment:
This is what tells main() that the async call has completed?

Reply:
Correct, if you noticed, we are using ConfigureAwait (false) that means that we leave the main thread, but the method needs to be executed in that one, so we need to call it from the main thread to work.